### PR TITLE
fix(#8902): external contributors can't access docker images

### PR DESCRIPTION
**Pull request by garethbowen**
# Description

#8902

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# Compose URLs
If Build CI hasn't passed, these may 404:

* [Core](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8902-fix-external-contributions/docker-compose/cht-core.yml)
* [CouchDB Single](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8902-fix-external-contributions/docker-compose/cht-couchdb.yml)
* [CouchDB Cluster](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8902-fix-external-contributions/docker-compose/cht-couchdb-clustered.yml)

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -4,11 +4,10 @@ on: [push, pull_request]
 
 env:
   COUCH_URL: http://admin:pass@localhost:5984/medic-test
-  BUILDS_SERVER: ${{ secrets.AUTH_MARKET_URL && '_couch/builds_testing' || '_couch/builds_external' }}
-  STAGING_SERVER: ${{ secrets.AUTH_MARKET_URL && '_couch/builds_4' || '_couch/builds_external' }}
+  BUILDS_SERVER: ${{ '_couch/builds_external' }}
+  STAGING_SERVER: ${{ '_couch/builds_external' }}
```
Comment (dianabarsan) [inline .github/workflows/build.yml:8]: This would mean that we never publish builds as we did before, right?

Comment (garethbowen) [reply .github/workflows/build.yml:8]: Please ignore that... the first commit was to turn myself into an external contributor so I could reproduce the issue - it'll be reverted before review!

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -176,7 +176,7 @@ jobs:
         node-version: ${{ env.NODE_VERSION }}
     - uses: actions/checkout@v4
     - name: Download docker images artifacts
-      uses: actions/download-artifact@v3
+      uses: actions/download-artifact@v4
```
Comment (garethbowen) [inline .github/workflows/build.yml:179]: If this version is out of sync with the "upload-artifact" version then the command fails quietly.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -277,7 +277,7 @@ jobs:
     steps:
     - name: Delete image artifacts
       if: ${{ !env.INTERNAL_CONTRIBUTOR }}
-      uses: geekyeggo/delete-artifact@v1
+      uses: geekyeggo/delete-artifact@v4
```
Comment (garethbowen) [inline .github/workflows/build.yml:280]: If this version is out of sync with the "upload-artifact" version then the command fails quietly.

**Comment by garethbowen** [comment]
@dianabarsan I think this is read for review now. There are two parts...

1. The artifact plugins all have to be compatible versions, in this case @v4. Before this change the artifact was not found.
2. Disabling the publish and upgrade testing steps which would take quite a lot of work to pull off.

**Review by dianabarsan — APPROVED** [review]
Cool! Thanks for figuring this out!

**Pull request merged**